internal/database/badger/database.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+// Package badger provides a database implementation using badger v3.
+package badger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/internal/database"
+	"github.com/dgraph-io/badger/v3"
+)
+
+// Database is database implementation using a badger/v3 database.
+type Database struct {
+	badgerDatabase *badger.DB
+}
+
+// New returns a new database based on a badger v3 database.
+func New(settings Settings) (database *Database, err error) {
```

Comment:
nit:
```suggestion
func NewDatabase(settings Settings) (database *Database, err error) {
```

Reply:
I think it's fine as `New` since the package is only dedicated to the badger database (and also the signature can be checked). A bit like `log.New` instead of `log.NewLogger`